rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -896,10 +1168,12 @@ extern "C"
 
         waitset->fixed_guard_conditions = fixed_guard_conditions;
         if (fixed_guard_conditions) {
+/*
```

Comment:
The check was too strict in my opinion. @richiprosima, what do you think of removing this check? In rclpy, we pass 0 as the number of guard conditions right now, which made this fail, is that a valid value?